test_launch_ros/test/test_launch_ros/test_normalize_parameters.py

```diff
@@ -230,3 +230,55 @@ def test_mixed_path_dicts():
     norm = normalize_parameters(orig)
     expected = (pathlib.Path('/foo/bar'), {'fiz.buz': 3}, pathlib.Path('/tmp/baz'))
     assert evaluate_parameters(LaunchContext(), norm) == expected
+
+
+def test_yaml_list_in_substitutions():
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text="['asd', 'fds', 'bsd']")}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': ('asd', 'fds', 'bsd')},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
+
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text='[1, 2, 3]')}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': (1, 2, 3)},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
+
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text='[1., 2., 3.]')}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': (1., 2., 3.)},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
+
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text='[True, 2., 3]')}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': '[True, 2., 3]'},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
+
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text='[(2, 4, 5), (2, 4, 5), (2, 4, 5)]')}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': '[(2, 4, 5), (2, 4, 5), (2, 4, 5)]'},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
+
+
+def test_unallowed_yaml_types_in_substitutions():
+    orig = [{'foo': 1, 'fiz': TextSubstitution(text="{'asd': 3}")}]
+    norm = normalize_parameters(orig)
+    expected = ({'foo': 1, 'fiz': "{'asd': 3}"},)
+    assert evaluate_parameters(LaunchContext(), norm) == expected
```

Comment:
I guess this is the kind of input you'd like to have an exception for but didn't add?

Reply:
I would like to raise an error in all this cases:
https://github.com/ros2/launch_ros/blob/1456aca92b837a4970e986abd2651f6316024404/test_launch_ros/test/test_launch_ros/test_normalize_parameters.py#L251-L284

Reply:
Started raising some exceptions in https://github.com/ros2/launch_ros/commit/4efbfe5e0933a3747954f78b1c765348711375e9.